src/receipt/saga.js

```diff
@@ -1,26 +1,33 @@
 import { call, put, takeEvery } from 'redux-saga/effects';
+import { history } from '@edx/frontend-platform';
+import { logError } from '@edx/frontend-platform/logging';
 
 // Actions
 import {
   FETCH_ORDER,
   fetchOrderBegin,
   fetchOrderSuccess,
+  fetchOrderFailure,
   fetchOrderReset,
 } from './actions';
 
 // Services
 import * as OrderApiService from './service';
 
 export function* handleFetchOrder(action) {
-  console.log('[Saga] 1.a) action', action);
-  const { orderToFetch } = action.payload;
-  // JK TODO: try and finally block? Remove console.log
-  console.log('[Saga] 1.b) orderToFetch:', orderToFetch);
-  yield put(fetchOrderBegin());
-  const result = yield call(OrderApiService.getOrder, orderToFetch);
-  console.log('[Saga] 1.c) result:', result);
-  yield put(fetchOrderSuccess(result));
-  console.log('[Saga] 1.d) success');
+  try {
+    const { orderToFetch } = action.payload;
+    yield put(fetchOrderBegin());
+    const result = yield call(OrderApiService.getOrder, orderToFetch);
+    yield put(fetchOrderSuccess(result));
+  } catch (error) {
```

Comment:
^ are there pieces that can move out of the try and into the else block, so it's clearer which bit we're targetting? Or could any of these fail separately/differently?